fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/FoldConstantRuleOnBE.java

```diff
@@ -486,9 +486,27 @@ public static List<Literal> getResultExpression(DataType type, PValues resultCon
             }
         } else if (type.isStringLikeType()) {
             int num = resultContent.getStringValueCount();
-            for (int i = 0; i < num; ++i) {
-                Literal literal = new StringLiteral(resultContent.getStringValue(i));
-                res.add(literal);
+
+            if (num > 0) {
+                for (int i = 0; i < num; ++i) {
+                    ByteString bytesValues = resultContent.getBytesValue(i);
+                    String stringValue = bytesValues.toStringUtf8();
+                    if ("\\N".equalsIgnoreCase(stringValue) && resultContent.hasHasNull()) {
+                        res.add(new NullLiteral(type));
+                    } else {
+                        res.add(new StringLiteral(stringValue));
+                    }
+                }
+            } else {
+                num = resultContent.getStringValueCount();
+                for (int i = 0; i < num; ++i) {
+                    String stringValue = resultContent.getStringValue(i);
+                    if ("\\N".equalsIgnoreCase(stringValue) && resultContent.hasHasNull()) {
```

Comment:
we could extract this replicate dealing of string to a seperate function to reduce codes